common/test/config.go

```diff
@@ -143,7 +143,9 @@ func NewMatrixKey(matrixKeyPath string) (err error) {
 	}
 
 	defer (func() {
-		err = keyOut.Close()
+		if err == nil {
```

Comment:
Again, `keyOut.Close` won't be called if `err != nil`. To fix that, you need to call `keyOut.Close()` first outside the `if`, then see what to do with the return value.

This also applies to the remaining occurrences.

Reply:
if an error occurs by calling this method i would guess there is no need to close keyout.